set.js

```diff
@@ -46,7 +47,7 @@ ProtoSet.prototype.MutantSet = function (listener) {
 }
 
 ProtoSet.prototype.add = function (valueOrObs) {
-  if (!~this.sources.indexOf(valueOrObs)) {
+  if (this.comparer ? this.sources.filter(x => this.comparer(x, valueOrObs)).length == 0 : !~this.sources.indexOf(valueOrObs)) {
```

Comment:
This is not the only occurrence of the check. There are quite a few other places that do this. Search for 
`~this.sources.indexOf` and `~self.sources.indexOf`.

What I would recommend is adding a `hasSource` method to `ProtoSet.prototype` that uses the new comparer function, and then use that method everywhere else.

```js
ProtoSet.prototype.hasSource = function (valueOrObs) {
  return this.comparer 
    ? this.sources.some(function (source) { return this.comparer(source, valueOrObs) }) 
    : this.sources.include(valueOrObs)
}
```

Note that I am using `array.some` here (I think this is a better choice than filtering) and not using the ES6 function syntax for [compatibility](https://github.com/mmckegg/mutant#compatibility) reasons.

Some examples of `hasSource` in use:

```js
ProtoSet.prototype.add = function (valueOrObs) {
  if (!this.hasSource(valueOrObs)) {
    this.sources.push(valueOrObs)
    ...
  }
}
```

```js
ProtoSet.prototype.set = function (values) {
 ...
  if (Array.isArray(values)) {
    ...
    values.forEach(function (value) {
      if (!self.hasSource(value)) {
        changed = true
        self.sources.push(value)
      }
    })
  }
  ...
}
```